webapp/Utils.js

```diff
@@ -198,6 +233,47 @@ const Utils = {
 
   execPythonMessage,
   evalPythonMessage,
+
+  flatten (obj, path = '') {
+    if (!(obj instanceof Object)) return { [path.replace(/\.$/g, '')]: obj };
+
+    return Object.keys(obj)
+      .reduce((output, key) => (obj instanceof Array
```

Comment:
I like the use of reduce but be aware that it's really unefficient in js when used like this. Also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat may help

Reply:
that's existing code that I extracted to utils